### ISSUE TITLE
Missing parent call when overriding traits methods

### ISSUE THREAD

**Issue by henriquecm** [hammer-tools/php-hammer#18]
For trait methods, the call `parent::method` doesn't exist.

So in this case we need to alias the method, like:

```php
use Impersonate {
    impersonatable as baseImpersonatable;
}
```

And on the override we call it like this.

```php
public function impersonatable(ImpersonateAuthorization $authorization): void
{
    $this->baseImpersonatable($authorization);
    $authorization->impersonator(static function (self $user) {
        return $user->hasRole('super-admin');
    });
}
```

When I do this, PHP Hammer still complains about the parent call.

**Comment by rentalhost** [comment]
Thank you for reporting this issue. We would appreciate it if you could provide more details about the error you encountered and provide a minimal reproducible code example. This will help us investigate and address the problem more effectively. Please feel free to share any additional information or code snippets related to the error.

**Comment by rentalhost** [comment]
I'm revisiting this problem, and I might have a slightly better understanding of what you mean.

From what I've gathered, even though you've performed a 'rename,' in PHP, you're actually [**creating a copy**](https://3v4l.org/cY03N#v8.2.11) of the 'impersonatable' method (which, within your sample class, can be called as 'impersonatable' or 'baseImpersonatable' as long as it hasn't been overridden by the class or a parent class up to the level where the trait was used).

Hence, the suggestion to call `parent::impersonatable()` is still valid because it exists and is implicitly declared for PHP. (Personally, I think it's confusing, but there's not much we can do in this case unless it's a PHP bug.)

I'm closing this issue since you haven't provided any further feedback. However, please feel free to reopen it if I've misunderstood your request, and provide a complete example if possible. Once I can reproduce it, I can consider further possibilities.

**Comment by henriquecm** [comment]
> From what I've gathered, even though you've performed a 'rename,' in PHP, you're actually [creating a copy](https://3v4l.org/cY03N#v8.2.11) of the 'impersonatable' method (which, within your sample class, can be called as 'impersonatable' or 'baseImpersonatable' as long as it hasn't been overridden by the class or a parent class up to the level where the trait was used).

This is not correct because trait add methods like they are in class and not on parent, they are more like an extension than an inheritance. And I am not creating a copy, I am aliasing the method.
https://www.php.net/manual/en/language.oop5.traits.php#language.oop5.traits.conflict

To expand the example, in this case Authenticatable is the parent.
```php
use Illuminate\Foundation\Auth\User as Authenticatable;
use Octopy\Impersonate\Concerns\Impersonate;
use Octopy\Impersonate\ImpersonateAuthorization;

class User extends Authenticatable {

    use Impersonate {
        impersonatable as baseImpersonatable;
    }

    public function impersonatable(ImpersonateAuthorization $authorization): void
    {
        $this->baseImpersonatable($authorization);
        $authorization->impersonator(static function (self $user) {
            return $user->hasRole('super-admin');
        });
    }
}
```

If I change the code to what hammer is suggesting, like this:
```php
public function impersonatable(ImpersonateAuthorization $authorization): void
{
    parent::impersonatable($authorization);
    $authorization->impersonator(static function (self $user) {
        return $user->hasRole('super-admin');
    });
}
```
I get the following error, `Call to undefined method App\Models\User::impersonatable()`

![image](https://github.com/hammer-tools/php-hammer/assets/15136037/a204c0ca-0827-4aa4-91f0-a12ec4804528)

In your code example, you are putting the trait in the parent class, that's why it both methods work.

I modified to look more like what is happening to me. And I get the same error.

https://3v4l.org/e3CDW#v8.2.11

**Comment by rentalhost** [comment]
Ah, now I get it!

I actually solved a similar problem, but not exactly this one because I hadn't fully understood it. I now realize that the issue is with the `parent` suggestion within the class that uses the trait itself, rather than in a more distant child class.

I'll address this issue very soon.

**Comment by rentalhost** [comment]
Thank you for your time in reporting this issue. The fix has been implemented in **version 0.25.1** and is **available exclusively** through the [**Releases**](https://github.com/hammer-tools/php-hammer/releases) page. It will be implemented in the JetBrains Marketplace once [**this issue**](https://github.com/hammer-tools/php-hammer/issues/10) that are affecting some users are resolved (no ETA).

**In response to your report:**

- The issue has been identified and fixed. Now, the condition you mentioned, where the inspection was suggesting the use of `parent::call()` at the same level where the trait was used, will no longer be suggested by the inspection.

If possible, install the version manually, test it, and see if it resolved the mentioned issue.
